### ISSUE TITLE
Provide better diagnostics for when SSL libraries are not installed.

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Base-ModuleBuild#6]
HTTP::Tiny has an annoying way of handing exceptions and as a result AB::MB has for a long time had the very unhelpful "Internal Exception" error for any connection error.

Bonus points: the best way to get the SSL prereqs is to specify `https` as a protocol instead of `http`.  This hasn't been documented well unfortunately, so the diagnostic for an SSL failure should direct to a FAQ or issue which explains the best way for the maintainer to fix it.

**Comment by plicease** [comment]
As of version 0.032 (back when `AB::MB` was part of `Alien-Base`), `Alien::Base::ModuleBuild` will add the SSL prereqs automatically on a share install when you specify a protocol of `https`.  This is preferred over explicitly specifying `Net::SSLeay` and `IO::Socket::SSL` as prereqs because they are not usually needed for a system install.

tl;dr, change your `alien_repository` from

```perl
  alien_repository => {
    protocol => 'http',
    ....
  }
```

to 

```perl
  alien_repository => {
    protocol => 'https',
    ....
  }
```
